app/services/external_upload_manager.rb

```diff
@@ -31,6 +33,17 @@ def promote_to_upload!
     # variable as well to check.
     tempfile = nil
     should_download = external_size < DOWNLOAD_LIMIT
+
+    # We require that the file size is specified ahead of time, and compare
+    # it here to make sure that people are not uploading excessively large
+    # files to the external provider. If this happens, the user will be banned
+    # from uploading to the external provider for N minutes.
+    if external_size != external_upload_stub.filesize
+      external_upload_stub.created_by.ban_from_external_uploads!(ban_minutes: SIZE_MISMATCH_BAN_MINUTES)
+      Discourse.store.delete_file(external_upload_stub.key)
```

Comment:
Do we need to delete the stub here too since it is no longer a valid stub?

Reply:
Yes good point